aopy/analysis/base.py

```diff
@@ -1734,4 +1735,91 @@ def calc_confidence_interval_overlap(CI1, CI2):
     # Calculate overlap ratio
     overlap = (overlap_width / min(width1, width2))
     
-    return overlap
\ No newline at end of file
+    return overlap
+
+
+def calc_max_coh_and_lags(lags, lfp_array1, lfp_array2, frequency_range):
+    """
+    Calculate the maximum coherence and corresponding lags between pairs of channels
+    in two local field potential (LFP) arrays within a specified frequency range.
+    
+    Parameters:
+    - lags: List of time lags (in seconds) to apply to the channels in lfp_array1.
+    - lfp_array1: First LFP array (e.g., motor cortex data) with shape (n_channels, n_timepoints).
+    - lfp_array2: Second LFP array (e.g., pre-motor cortex data) with shape (n_channels, n_timepoints).
+    - frequency_range: Frequency range of interest (e.g., [0, 4] Hz for delta band).
+
+    Returns:
+    - max_coherence_matrix: Matrix of maximum coherence values for each channel pair.
+    - lag_of_max_coherence_matrix: Matrix of corresponding lags (in ms) for max coherence values.
+    """
+    # Taper parameters
+    NW, BW, step, fk, fs = 0.5, 4, 0.005, 200, 2500
+    n, half_bw, num_tapers = precondition.convert_taper_parameters(NW, BW)
+    print(f"Using {num_tapers} tapers, taper length {n}, half-bandwidth {half_bw}")
+
+    # Get channel counts
+    num_channels_lfp1 = lfp_array1.shape[0]
+    num_channels_lfp2 = lfp_array2.shape[0]
+
+    # Initialize matrices for max coherence values and their corresponding lags
+    max_coherence_matrix = np.zeros((num_channels_lfp1, num_channels_lfp2))
+    lag_of_max_coherence_matrix = np.zeros((num_channels_lfp1, num_channels_lfp2))
+
```

Comment:
Can you initialize these arrays as np.nan values to make it obvious if the answer is 0 or if the value just didn't get filled in? You can just add "*np.nan" to the end of each of these lines.